api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -513,29 +517,48 @@ private FirecloudWorkspaceId getFcBillingProject(DbAccessTier accessTier, Worksp
     }
     final ImmutableMap<String, WorkspaceAccessLevel> aclsByEmail = shareRolesMapBuilder.build();
 
+    List<String> usersHavePermission = new ArrayList<>();
+    // Revoke lifesciene permission before asking Firecloud to remove users. Because after unshare
```

Comment:
I think this is probably the best way forward, but it is getting pretty complicated. Do we have a spike ticket, or CI ticket for pushing either the lifescience role down into Terra for proper support, or adding better support for getting a user's pet SA?